app/grandchallenge/reader_studies/views.py

```diff
@@ -867,6 +935,91 @@ def ground_truth(self, request, pk=None, case_pk=None):
         )
 
 
+class DisplaySetViewSet(
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.ListModelMixin,
+    GenericViewSet,
+):
+    serializer_class = DisplaySetSerializer
+    permission_classes = [DjangoObjectPermissions]
+    filter_backends = [DjangoFilterBackend, ObjectPermissionsFilter]
+    filterset_fields = ["reader_study"]
+    renderer_classes = (
+        *api_settings.DEFAULT_RENDERER_CLASSES,
+        PaginatedCSVRenderer,
+    )
+
+    @property
+    def reader_study(self):
+        reader_study_pk = self.request.query_params.get("reader_study")
+        if reader_study_pk:
+            return ReaderStudy.objects.get(pk=reader_study_pk)
+
+    def get_serializer_class(self):
+        if self.action in ["partial_update", "update", "create"]:
+            return DisplaySetPostSerializer
+        return DisplaySetSerializer
+
+    def partial_update(self, request, pk=None):
+        instance = self.get_object()
+        if request.data.get("value"):
+            civ = ComponentInterfaceValue.objects.get(
+                id=request.data.get("value")
+            )
+            civ.displays_set.clear()
```

Comment:
I'm having a hard time following this. Would be really handy to split this up into some methods so that the higher-level logic is clear. What is happening with clearing all display sets here? Could it be that the civ belongs to multiple display sets (eg from other reader studies, see the archive -> reader study link).

Reply:
I'm also having a hard time following this. Maybe a test would help? Maybe I missed it, but I only see a test that updates the `values` of a display_set, but not one that provides the id of a CIV as update argument?

Reply:
I added some comments and a test. Splitting it up into smaller functions made things a bit messy imo.